src/dmd/expressionsem.d

```diff
@@ -8208,7 +8209,10 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
         e.type = e.e2.type;
         if (e.type is Type.tvoid)
+        {
+            checkMustUse(e.e1, sc);
```

Comment:
Seems like this could be included in `discardValue` given that they deal with the same issue and are always called in direct succession.

Reply:
`discardValue` calls itself recursively, so doing this would result in `checkMustUse` being called on expressions that are ok to discard.

Reply:
Okay. Both function calls could also be joined into a seperate method (allthough that might be a bit excessive)